pipeline-templates/lint.yaml

```diff
@@ -2,17 +2,17 @@ parameters:
   pool: ''
 
 jobs:
-  - job: TSLint
+  - job: ESLint
     pool:
       ${{ if eq(parameters.useOneEngineeringPool, 'true') }}:
         ${{ parameters.pool }}
       ${{ else }}:
         vmImage: ${{ parameters.pool.vmImage }}
-    displayName: '🧶 TSLint'
+    displayName: '🧶 ESLint'
     steps:
     - template: install-node.yaml
+    - script: build.cmd
     - bash: |
-        npm install --cache /temp/empty-cache
-        npm install tslint --reg https://registry.npmjs.org/ --verbose
+        npm install eslint@^8.57.0 @typescript-eslint/parser@^8.0.0 @typescript-eslint/eslint-plugin@^8.0.0 --reg https://registry.npmjs.org/ --save-dev --verbose

```

Comment:
How often would these versions need to change?

Reply:
The `^` will allow them to change automatically, up to the latest major. 8 is the latest major that supports the things we need in typescript, (9 being the true latest) so this should be good for a while. Eventually it will need to be changed to 9, but at this point it will require a lot of work to adjust to breaking changes like this PR did.